File: 3088-make-string-anti-palindrome/3088-make-string-anti-palindrome.py
class Solution(object):
    def makeAntiPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        n = len(s)
        res = sorted(c for c in s)
        
        i = j = (n+1)//2
        while j < n and res[j] == res[i]:
            j += 1
        while res[i] == res[n-i-1]:
            if j == n: return "-1"
            res[i], res[j] = res[j], res[i]
            j += 1
            i += 1
        return "".join(res)

File: 3088-make-string-anti-palindrome/3088-make-string-anti-palindrome.cpp
class Solution {
public:
    string makeAntiPalindrome(string s) {
        int n = s.length();
        int i = (n+1)/2; int j = (n+1)/2;
        sort(s.begin(), s.end());
        while(j < n && s[i] == s[j]) ++j;
        while(s[i] == s[n-i-1]) {
            if(j == n) return "-1";
            swap(s[i++], s[j++]);
        }
        return s;
    }
};